File: CandyRainTolstonogova/CandyRainC++/CandyRain/Sweet.cpp
#include "Sweet.h"
#include <iostream> // For cout
#include <cstdlib> // For abs()
#include <string>
#include "GameMain.h"
Sweet::Sweet(int x, int y,  std::string spritepath, int spd, int num, bool ptouch, bool ftouch) :
    FallingObject(x, y, spritepath, spd, num, ptouch, ftouch) {
    std::cout << "ќбъект Sweet создан" << std::endl;
}
Sweet::~Sweet() {
    std::cout << "ќбъект Sweet уничтожен" << std::endl;
}
void Sweet::PLayerCatchSweet(int playerX, int playerY){
    if (abs(GetpositionX() - playerX) < GameMain::playersize && abs(GetpositionY() - playerY) < GameMain::playersize) {
        Setplayertouch (true) ;
        std::cout << "Sweet пойман игроком!" << std::endl;
    }
}

void Sweet::SweetTouchFloor() {
    if (GetpositionY() <= GameMain::floorY) {
        Setfloortouch (true);
        std::cout << "Sweet коснулс€ пола" << std::endl;
    }
} 
void Sweet::ChangeCandy(int& countcandy) {
    countcandy-= GameMain::countchange;
    std::cout << "—четчик конфет уменьшен: " << countcandy << std::endl;
}

void Sweet::ChangeCake(int& countcake) {
    countcake-= GameMain::countchange;
    std::cout << "—четчик тортов уменьшен: " << countcake << std::endl;
}

void Sweet::ChangeDonut(int& countdonut) {
    countdonut-= GameMain::countchange;
    std::cout << "—четчик пончиков уменьшен: " << countdonut << std::endl;
}

File: CandyRainTolstonogova/CandyRainJava/Sweet.java
public class Sweet extends FallingObject {
    
    public Sweet(int x, int y, String sprite, int spd, int num, boolean pTouch, boolean fTouch) {
        super(x, y, sprite, spd, num, pTouch, fTouch);
        System.out.println("Объект Sweet создан");
    }
    
    public void playerCatchSweet(int playerX, int playerY) {
        if (Math.abs(getPositionX() - playerX) < GameMain.PLAYER_SIZE && 
            Math.abs(getPositionY() - playerY) < GameMain.PLAYER_SIZE) {
            setPlayerTouch(true);
            System.out.println("Sweet пойман игроком!");
        }
    }
    
    public void sweetTouchFloor() {
        if (getPositionY() <= GameMain.FLOOR_Y) {
            setFloorTouch(true);
            System.out.println("Sweet коснулся пола");
        }
    }
    
    public void changeCandy(SweetsDistribution distribution) {
        distribution.setCountCandy(distribution.getCountCandy() - GameMain.COUNT_CHANGE);
        System.out.println("Счетчик конфет уменьшен: " + distribution.getCountCandy());
    }
    
    public void changeCake(SweetsDistribution distribution) {
        distribution.setCountCake(distribution.getCountCake() - GameMain.COUNT_CHANGE);
        System.out.println("Счетчик тортов уменьшен: " + distribution.getCountCake());
    }
    
    public void changeDonut(SweetsDistribution distribution) {
        distribution.setCountDonut(distribution.getCountDonut() - GameMain.COUNT_CHANGE);
        System.out.println("Счетчик пончиков уменьшен: " + distribution.getCountDonut());
    }
}